public/app/features/trails/MetricOverviewScene.tsx

```diff
@@ -0,0 +1,232 @@
+import { getDatasourceSrv } from '../plugins/datasource_srv';
+import { ALL_VARIABLE_VALUE } from '../variables/constants';
+
+import { DataTrail } from './DataTrail';
+import { MetricScene } from './MetricScene';
+import { trailDS, VAR_DATASOURCE, VAR_FILTERS, VAR_FILTERS_EXPR, VAR_GROUP_BY, VAR_METRIC_EXPR } from './shared';
+import { getMetricSceneFor } from './utils';
+export interface MetricOverviewSceneState extends SceneObjectState {
+  labels: Array<SelectableValue<string>>;
+  metadata?: PromMetricsMetadataItem;
+  languageProvider?: PrometheusLanguageProvider;
+  loading?: boolean;
+}
+
+export class MetricOverviewScene extends SceneObjectBase<MetricOverviewSceneState> {
+  constructor(state: Partial<MetricOverviewSceneState>) {
+    super({
+      $variables: state.$variables ?? getVariableSet(),
+      labels: state.labels ?? [],
+      ...state,
+    });
+
+    this.addActivationHandler(this._onActivate.bind(this));
+  }
+
+  private getVariable(): QueryVariable {
+    const variable = sceneGraph.lookupVariable(VAR_GROUP_BY, this)!;
+    if (!(variable instanceof QueryVariable)) {
+      throw new Error('Group by variable not found');
+    }
+
+    return variable;
+  }
+
+  private _onActivate() {
+    const metricScene = getMetricSceneFor(this);
+    this.updateMetadata();
+    this.updateLanguageProvider();
+
+    metricScene.subscribeToState((newState, oldState) => {
+      if (newState.metric !== oldState.metric) {
+        this.updateMetadata();
+      }
+    });
+
+    const variable = this.getVariable();
+    variable.subscribeToState((newState, oldState) => {
+      if (
+        newState.options !== oldState.options ||
+        newState.value !== oldState.value ||
+        newState.loading !== oldState.loading
+      ) {
+        const labels = this.getLabelOptions(this.getVariable()).filter((l) => l.value !== ALL_VARIABLE_VALUE);
+        this.setState({ labels, loading: variable.state.loading });
+        this.updateLabelValuesCount();
+      }
+    });
+
+    this.setState({ loading: variable.state.loading });
+  }
+
+  private updateLanguageProvider() {
+    const dsUid = sceneGraph.getAncestor(this, DataTrail).state.$variables?.getByName(VAR_DATASOURCE)?.getValue();
+    if (typeof dsUid === 'string') {
+      getDatasourceSrv()
+        .get(dsUid)
+        .then((ds) => {
+          const languageProvider: PrometheusLanguageProvider = ds.languageProvider;
+          this.setState({ languageProvider });
+        });
+    }
+  }
+
+  private updateMetadata() {
+    const metricScene = getMetricSceneFor(this);
+    const metric = metricScene.state.metric;
+    const dsUid = sceneGraph.getAncestor(this, DataTrail).state.$variables?.getByName(VAR_DATASOURCE)?.getValue();
+    if (typeof dsUid === 'string') {
+      getDatasourceSrv()
+        .get(dsUid)
+        .then((ds) => {
+          const langProvider: PrometheusLanguageProvider = ds.languageProvider;
```

Comment:
We should try to refactor this so we can share the same language provider here as the one fetched in updateLanguageProvider